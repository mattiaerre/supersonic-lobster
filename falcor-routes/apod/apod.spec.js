require('babel-core/register');
require('babel-polyfill');
const { APOD, COPYRIGHT, DATE, EXPLANATION, HDURL, MEDIA_TYPE, TITLE, URL } = require('./fields');
const apod = require('./apod');
const image = require('./sample-image');
const video = require('./sample-video');

const samples = [{ type: 'image', data: image }, { type: 'video', data: video }];


const route = apod('TEST_KEY')[0];

describe('happy path', () => {
  samples.forEach((sample) => {
    describe(`sample data "${sample.type}"`, () => {
      fetch.mockResponse(JSON.stringify(sample.data));

      test('route.route to match snapshot', () => {
        expect(route.route).toMatchSnapshot();
      });

      [DATE, COPYRIGHT, EXPLANATION, HDURL, MEDIA_TYPE, TITLE, URL].forEach((field) => {
        test(`route.route to contain "${field}"`, () => {
          expect(route.route).toContain(field);
        });

        test(`route.get(${JSON.stringify([APOD, [field]])}) to match snapshot`, async () => {
          const results = await route.get([APOD, [field]]);
          expect(results).toMatchSnapshot();
        });

        test(`route.get(${JSON.stringify([APOD, [field]])}) to be defined`, async () => {
          const results = await route.get([APOD, [field]]);
          expect(results[0].value).toBeDefined();
        });
      });
    });
  });
});

describe('special case', () => {
  test(`route.get(${JSON.stringify([APOD, [URL]])}) not to contain HTTP protocol`, async () => {
    const results = await route.get([APOD, [URL]]);
    expect(results[0].value).not.toContain('http:');
    expect(results[0].value).not.toContain('https:');
  });
});

describe('exception', () => {
  const BAD_FIELD = 'bad_field';
  test(`route.get(${JSON.stringify([APOD, [BAD_FIELD]])}) to match snapshot`, async () => {
    const results = await route.get([APOD, [BAD_FIELD]]);
    expect(results).toMatchSnapshot();
  });

  test(`route.get(${JSON.stringify([APOD, [DATE]])}) to match snapshot`, async () => {
    fetch.mockResponse('I\'m not a JSON');
    const results = await route.get([APOD, [DATE]]);
    expect(results).toMatchSnapshot();
  });
});
