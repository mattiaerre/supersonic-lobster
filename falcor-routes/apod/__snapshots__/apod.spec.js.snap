exports[`exception route.get(["apod",["bad_field"]]) to match snapshot 1`] = `Array []`;

exports[`exception route.get(["apod",["date"]]) to match snapshot 1`] = `Array []`;

exports[`happy path sample data "image" route.get(["apod",["copyright","date","explanation","hdurl","media_type","title","url"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "copyright",
    ],
    "value": "Yuri Beletsky",
  },
  Object {
    "path": Array [
      "apod",
      "date",
    ],
    "value": "2017-02-21",
  },
  Object {
    "path": Array [
      "apod",
      "explanation",
    ],
    "value": "The night sky is always changing.  Featured here are changes that occurred over a six hour period in late 2014 June behind the dual 6.5-meter Magellan Telescopes at Las Campanas Observatory in Chile. The initial red glow on the horizon is airglow, a slight cooling of high air by the emission of specific colors of light. Bands of airglow are also visible throughout the time-lapse video. Early in the night, car headlights flash on the far left.  Satellites quickly shoot past as they circle the Earth and reflect sunlight. A long and thin cloud passes slowly overhead.  The Large Magellanic Cloud rises on the left, while the expansive central band of our Milky Way Galaxy arches and pivots as the Earth rotates.  As the night progresses, the Magellan telescopes swivel and stare as they explore pre-determined patches of the night sky. Every night, every sky changes differently, even though the phenomena at play are usually the same.   Explore the Universe: Random APOD Generator",
  },
  Object {
    "path": Array [
      "apod",
      "hdurl",
    ],
    "value": "",
  },
  Object {
    "path": Array [
      "apod",
      "media_type",
    ],
    "value": "video",
  },
  Object {
    "path": Array [
      "apod",
      "title",
    ],
    "value": "An Active Night over the Magellan Telescopes",
  },
  Object {
    "path": Array [
      "apod",
      "url",
    ],
    "value": "//www.youtube.com/embed/hkLRmQC7fR4?rel=0&showinfo=0",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["copyright"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "copyright",
    ],
    "value": "Yuri Beletsky",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["date"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "date",
    ],
    "value": "2017-02-21",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["explanation"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "explanation",
    ],
    "value": "The night sky is always changing.  Featured here are changes that occurred over a six hour period in late 2014 June behind the dual 6.5-meter Magellan Telescopes at Las Campanas Observatory in Chile. The initial red glow on the horizon is airglow, a slight cooling of high air by the emission of specific colors of light. Bands of airglow are also visible throughout the time-lapse video. Early in the night, car headlights flash on the far left.  Satellites quickly shoot past as they circle the Earth and reflect sunlight. A long and thin cloud passes slowly overhead.  The Large Magellanic Cloud rises on the left, while the expansive central band of our Milky Way Galaxy arches and pivots as the Earth rotates.  As the night progresses, the Magellan telescopes swivel and stare as they explore pre-determined patches of the night sky. Every night, every sky changes differently, even though the phenomena at play are usually the same.   Explore the Universe: Random APOD Generator",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["hdurl"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "hdurl",
    ],
    "value": "",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["media_type"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "media_type",
    ],
    "value": "video",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["title"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "title",
    ],
    "value": "An Active Night over the Magellan Telescopes",
  },
]
`;

exports[`happy path sample data "image" route.get(["apod",["url"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "url",
    ],
    "value": "//www.youtube.com/embed/hkLRmQC7fR4?rel=0&showinfo=0",
  },
]
`;

exports[`happy path sample data "image" route.route to match snapshot 1`] = `"apod[\"date\", \"copyright\", \"explanation\", \"hdurl\", \"media_type\", \"title\", \"url\"]"`;

exports[`happy path sample data "video" route.get(["apod",["copyright","date","explanation","hdurl","media_type","title","url"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "copyright",
    ],
    "value": "Yuri Beletsky",
  },
  Object {
    "path": Array [
      "apod",
      "date",
    ],
    "value": "2017-02-21",
  },
  Object {
    "path": Array [
      "apod",
      "explanation",
    ],
    "value": "The night sky is always changing.  Featured here are changes that occurred over a six hour period in late 2014 June behind the dual 6.5-meter Magellan Telescopes at Las Campanas Observatory in Chile. The initial red glow on the horizon is airglow, a slight cooling of high air by the emission of specific colors of light. Bands of airglow are also visible throughout the time-lapse video. Early in the night, car headlights flash on the far left.  Satellites quickly shoot past as they circle the Earth and reflect sunlight. A long and thin cloud passes slowly overhead.  The Large Magellanic Cloud rises on the left, while the expansive central band of our Milky Way Galaxy arches and pivots as the Earth rotates.  As the night progresses, the Magellan telescopes swivel and stare as they explore pre-determined patches of the night sky. Every night, every sky changes differently, even though the phenomena at play are usually the same.   Explore the Universe: Random APOD Generator",
  },
  Object {
    "path": Array [
      "apod",
      "hdurl",
    ],
    "value": "",
  },
  Object {
    "path": Array [
      "apod",
      "media_type",
    ],
    "value": "video",
  },
  Object {
    "path": Array [
      "apod",
      "title",
    ],
    "value": "An Active Night over the Magellan Telescopes",
  },
  Object {
    "path": Array [
      "apod",
      "url",
    ],
    "value": "//www.youtube.com/embed/hkLRmQC7fR4?rel=0&showinfo=0",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["copyright"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "copyright",
    ],
    "value": "Yuri Beletsky",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["date"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "date",
    ],
    "value": "2017-02-21",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["explanation"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "explanation",
    ],
    "value": "The night sky is always changing.  Featured here are changes that occurred over a six hour period in late 2014 June behind the dual 6.5-meter Magellan Telescopes at Las Campanas Observatory in Chile. The initial red glow on the horizon is airglow, a slight cooling of high air by the emission of specific colors of light. Bands of airglow are also visible throughout the time-lapse video. Early in the night, car headlights flash on the far left.  Satellites quickly shoot past as they circle the Earth and reflect sunlight. A long and thin cloud passes slowly overhead.  The Large Magellanic Cloud rises on the left, while the expansive central band of our Milky Way Galaxy arches and pivots as the Earth rotates.  As the night progresses, the Magellan telescopes swivel and stare as they explore pre-determined patches of the night sky. Every night, every sky changes differently, even though the phenomena at play are usually the same.   Explore the Universe: Random APOD Generator",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["hdurl"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "hdurl",
    ],
    "value": "",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["media_type"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "media_type",
    ],
    "value": "video",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["title"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "title",
    ],
    "value": "An Active Night over the Magellan Telescopes",
  },
]
`;

exports[`happy path sample data "video" route.get(["apod",["url"]]) to match snapshot 1`] = `
Array [
  Object {
    "path": Array [
      "apod",
      "url",
    ],
    "value": "//www.youtube.com/embed/hkLRmQC7fR4?rel=0&showinfo=0",
  },
]
`;

exports[`happy path sample data "video" route.route to match snapshot 1`] = `"apod[\"date\", \"copyright\", \"explanation\", \"hdurl\", \"media_type\", \"title\", \"url\"]"`;
